hollow/src/main/java/com/netflix/hollow/api/producer/AbstractHollowProducer.java

```diff
@@ -702,7 +702,9 @@ private ReadStateHelper checkIntegrity(
                     if (!reverseChecksum.equals(currentChecksum)) {
                         throw new HollowProducer.ChecksumValidationException(HollowProducer.Blob.Type.REVERSE_DELTA);
                     }
-                    result = readStates.swap();
+                    // undo the changes we made to the read states
```

Comment:
Deltas for large datasets can take a relatively long time to apply.  It seems wasteful to do this every time.

There must be a better way to accomplish this -- e.g. detect when the schemas differ, then apply this logic selectively in those rare cases.

Reply:
Hm, yeah comparing all the object schemas would work. I'll probably have to add hashCode to make the comparisons easier but it's probably worth it.